File: tests/rosetta/transpiler/php/abc-problem.php
<?php
ini_set('memory_limit', '-1');
$now_seed = 0;
$now_seeded = false;
$s = getenv('MOCHI_NOW_SEED');
if ($s !== false && $s !== '') {
    $now_seed = intval($s);
    $now_seeded = true;
}
function _now() {
    global $now_seed, $now_seeded;
    if ($now_seeded) {
        $now_seed = ($now_seed * 1664525 + 1013904223) % 2147483647;
        return $now_seed;
    }
    return hrtime(true);
}
function _str($x) {
    if (is_array($x)) {
        $isList = array_keys($x) === range(0, count($x) - 1);
        if ($isList) {
            $parts = [];
            foreach ($x as $v) { $parts[] = _str($v); }
            return '[' . implode(' ', $parts) . ']';
        }
        $parts = [];
        foreach ($x as $k => $v) { $parts[] = _str($k) . ':' . _str($v); }
        return 'map[' . implode(' ', $parts) . ']';
    }
    if (is_bool($x)) return $x ? 'true' : 'false';
    if ($x === null) return 'null';
    return strval($x);
}
$__start_mem = memory_get_usage();
$__start = _now();
  function fields($s) {
  $res = [];
  $cur = '';
  $i = 0;
  while ($i < strlen($s)) {
  $c = substr($s, $i, $i + 1 - $i);
  if ($c == ' ') {
  if (strlen($cur) > 0) {
  $res = array_merge($res, [$cur]);
  $cur = '';
};
} else {
  $cur = $cur . $c;
}
  $i = $i + 1;
};
  if (strlen($cur) > 0) {
  $res = array_merge($res, [$cur]);
}
  return $res;
};
  function canSpell($word, $blks) {
  if (strlen($word) == 0) {
  return true;
}
  $c = strtolower(substr($word, 0, 1 - 0));
  $i = 0;
  while ($i < count($blks)) {
  $b = $blks[$i];
  if ($c == strtolower(substr($b, 0, 1 - 0)) || $c == strtolower(substr($b, 1, 2 - 1))) {
  $rest = [];
  $j = 0;
  while ($j < count($blks)) {
  if ($j != $i) {
  $rest = array_merge($rest, [$blks[$j]]);
}
  $j = $j + 1;
};
  if (canSpell(substr($word, 1), $rest)) {
  return true;
};
}
  $i = $i + 1;
};
  return false;
};
  function newSpeller($blocks) {
  $bl = fields($blocks);
  return function($w) use ($bl, $blocks) {
  return canSpell($w, $bl);
};
};
  function main() {
  $sp = newSpeller('BO XK DQ CP NA GT RE TG QD FS JW HU VI AN OB ER FS LY PC ZM');
  foreach (['A', 'BARK', 'BOOK', 'TREAT', 'COMMON', 'SQUAD', 'CONFUSE'] as $word) {
  echo rtrim($word . ' ' . _str($sp($word))), PHP_EOL;
};
};
  main();
$__end = _now();
$__end_mem = memory_get_usage();
$__duration = intdiv($__end - $__start, 1000);
$__mem_diff = max(0, $__end_mem - $__start_mem);
$__bench = ["duration_us" => $__duration, "memory_bytes" => $__mem_diff, "name" => "main"];
$__j = json_encode($__bench, 128);
$__j = str_replace("    ", "  ", $__j);
echo $__j, PHP_EOL;;

File: tests/rosetta/transpiler/Clojure/abc-problem.clj
(ns main (:refer-clojure :exclude [fields canSpell newSpeller main]))

(require 'clojure.set)

(defn in [x coll]
  (cond (string? coll) (clojure.string/includes? coll x) (map? coll) (contains? coll x) (sequential? coll) (some (fn [e] (= e x)) coll) :else false))

(def nowSeed (atom (let [s (System/getenv "MOCHI_NOW_SEED")] (if (and s (not (= s ""))) (Integer/parseInt s) 0))))

(declare fields canSpell newSpeller main)

(defn fields [s]
  (try (do (def res []) (def cur "") (def i 0) (while (< i (count s)) (do (def c (subs s i (+ i 1))) (if (= c " ") (when (> (count cur) 0) (do (def res (conj res cur)) (def cur ""))) (def cur (str cur c))) (def i (+ i 1)))) (when (> (count cur) 0) (def res (conj res cur))) (throw (ex-info "return" {:v res}))) (catch clojure.lang.ExceptionInfo e (if (= (ex-message e) "return") (get (ex-data e) :v) (throw e)))))

(defn canSpell [word blks]
  (try (do (when (= (count word) 0) (throw (ex-info "return" {:v true}))) (def c (clojure.string/lower-case (subs word 0 1))) (def i 0) (while (< i (count blks)) (do (def b (nth blks i)) (when (or (= c (clojure.string/lower-case (subs b 0 1))) (= c (clojure.string/lower-case (subs b 1 2)))) (do (def rest []) (def j 0) (while (< j (count blks)) (do (when (not= j i) (def rest (conj rest (nth blks j)))) (def j (+ j 1)))) (when (canSpell (subs word 1 (count word)) rest) (throw (ex-info "return" {:v true}))))) (def i (+ i 1)))) (throw (ex-info "return" {:v false}))) (catch clojure.lang.ExceptionInfo e (if (= (ex-message e) "return") (get (ex-data e) :v) (throw e)))))

(defn newSpeller [blocks]
  (try (do (def bl (fields blocks)) (throw (ex-info "return" {:v (fn [w] (canSpell w bl))}))) (catch clojure.lang.ExceptionInfo e (if (= (ex-message e) "return") (get (ex-data e) :v) (throw e)))))

(defn main []
  (do (def sp (newSpeller "BO XK DQ CP NA GT RE TG QD FS JW HU VI AN OB ER FS LY PC ZM")) (doseq [word ["A" "BARK" "BOOK" "TREAT" "COMMON" "SQUAD" "CONFUSE"]] (println (str (str word " ") (str (sp word)))))))

(defn -main []
  (let [rt (Runtime/getRuntime)
    start-mem (- (.totalMemory rt) (.freeMemory rt))
    start (System/nanoTime)]
      (main)
      (System/gc)
      (let [end (System/nanoTime)
        end-mem (- (.totalMemory rt) (.freeMemory rt))
        duration-us (quot (- end start) 1000)
        memory-bytes (Math/abs ^long (- end-mem start-mem))]
        (println (str "{\n  \"duration_us\": " duration-us ",\n  \"memory_bytes\": " memory-bytes ",\n  \"name\": \"main\"\n}"))
      )
    ))

(-main)
